### PR TITLE
Create surplis deficit chart

### PR DESCRIPTION
**Pull request by bobhaslett**

### CODE REVIEW

File: surplus-deficit/parseData.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        // Format the data that is used to draw highlight tonal bands
+        const boundaries = data.filter(d => (d.highlight === 'begin' || d.highlight === 'end'));
+        const highlights = [];
+
+        boundaries.forEach((d, i) => {
+            if (d.highlight === 'begin') {
+                highlights.push({ begin: d.date, end: boundaries[i + 1].date });
+            }
+        });
+        console.log(highlights);
```
Comment (aendrew) [inline surplus-deficit/parseData.js]: Remove extraneous console.log

File: surplus-deficit/parseData.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+/**
+ * Calculates the extent of multiple columns
+ * @param  {[type]} d       [description]
+ * @param  {[type]} columns [description]
+ * @param  {[type]} yMin    [description]
+ * @return {[type]}         [description]
+ */
+// a function that calculates the cumulative ma min values of the dataset
+function getMaxMin(values) {
```
Comment (aendrew) [inline surplus-deficit/parseData.js]: This function is neither used nor exported. Please remove if extraneous.

File: surplus-deficit/surplusDeficit.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+    let annotate = false; // eslint-disable-line
+    let colourScale = d3.scaleOrdinal();
+
+    function chart(parent) {
+        const width = window.plotDim.width;
+        const height = window.plotDim.height;
+
+        const area = d3.area()
+            .x(d => xScale(d.data.date))
+            .y0(d => yScale(yAxisHighlight))
```
Comment (aendrew) [inline surplus-deficit/surplusDeficit.js]: If `d` isn't used, don't pass it, use `()` instead.

File: surplus-deficit/surplusDeficit.js [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+            .append('path')
+            .attr('d', area)
+            .style('fill', 'white')
+            .attr('opacity', 0.5);
+
+        parent.append('rect')
+            .attr('x', 0)
+            .attr('y', 0)
+            .attr('width', width)
+            .attr('height', d => yScale(yAxisHighlight))
```
Comment (aendrew) [inline surplus-deficit/surplusDeficit.js]: If `d` isn't used, don't pass it, use `()` instead.

File: surplus-deficit/surplusDeficit.js [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+            .attr('fill', colourScale(3))
+            .attr('clip-path', `url(#${frameName}chartMask)`);
+
+        parent.append('rect')
+            .attr('x', 0)
+            .attr('y', d => yScale(yAxisHighlight))
```
Comment (aendrew) [inline surplus-deficit/surplusDeficit.js]: If `d` isn't used, don't pass it, use `()` instead.

File: surplus-deficit/surplusDeficit.js [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+            .attr('width', width)
+            .attr('height', d => height - yScale(yAxisHighlight))
```
Comment (aendrew) [inline surplus-deficit/surplusDeficit.js]: If `d` isn't used, don't pass it, use `()` instead.

**Comment by aendrew** [comment]
Added linting commit.

File: surplus-deficit/parseData.js (hunk 1 above)
Comment (bobhaslett) [reply surplus-deficit/parseData.js]: Done

File: surplus-deficit/parseData.js (hunk 2 above)
Comment (bobhaslett) [reply surplus-deficit/parseData.js]: Done

File: surplus-deficit/surplusDeficit.js (hunk 4 above)
Comment (bobhaslett) [reply surplus-deficit/surplusDeficit.js]: Done

File: surplus-deficit/surplusDeficit.js (hunk 5 above)
Comment (bobhaslett) [reply surplus-deficit/surplusDeficit.js]: Done

File: surplus-deficit/surplusDeficit.js (hunk 6 above)
Comment (bobhaslett) [reply surplus-deficit/surplusDeficit.js]: Done

**Pull request closed without merging**